libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -0,0 +1,801 @@
+                    }
+
+                    magnetic = new uavcan::Subscriber<uavcan::equipment::ahrs::MagneticFieldStrength>(*node);
+                    const int magnetic_start_res = magnetic->start(magnetic_cb);
+                    if (magnetic_start_res < 0) {
+                        debug_uavcan(1, "UAVCAN Compass subscriber start problem\n\r");
+                        return false;
+                    }
+
+                    air_data_sp = new uavcan::Subscriber<uavcan::equipment::air_data::StaticPressure>(*node);
+                    const int air_data_sp_start_res = air_data_sp->start(air_data_sp_cb);
+                    if (air_data_sp_start_res < 0) {
+                        debug_uavcan(1, "UAVCAN Baro subscriber start problem\n\r");
+                        return false;
+                    }
+
+                    air_data_st = new uavcan::Subscriber<uavcan::equipment::air_data::StaticTemperature>(*node);
+                    const int air_data_st_start_res = air_data_st->start(air_data_st_cb);
+                    if (air_data_st_start_res < 0) {
+                        debug_uavcan(1, "UAVCAN Temperature subscriber start problem\n\r");
+                        return false;
+                    }
+
+                    act_out_array = new uavcan::Publisher<uavcan::equipment::actuator::ArrayCommand>(*node);
+                    act_out_array->setTxTimeout(uavcan::MonotonicDuration::fromMSec(20));
+                    act_out_array->setPriority(uavcan::TransferPriority::OneLowerThanHighest);
+
+                    esc_raw = new uavcan::Publisher<uavcan::equipment::esc::RawCommand>(*node);
+                    esc_raw->setTxTimeout(uavcan::MonotonicDuration::fromMSec(20));
+                    esc_raw->setPriority(uavcan::TransferPriority::OneLowerThanHighest);
+
+                    /*
+                     * Informing other nodes that we're ready to work.
+                     * Default mode is INITIALIZING.
+                     */
+                    node->setModeOperational();
+
+                    _initialized = true;
+
+                    debug_uavcan(1, "UAVCAN: init done\n\r");
+
+                    return true;
+                }
+            }
+        }
+
+        if (_initialized == true) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void AP_UAVCAN::rc_out_sem_take()
+{
+    bool sem_ret = _rc_out_sem->take(0);
+    if (sem_ret == false) {
+        debug_uavcan(1, "AP_UAVCAN RCOut semaphore fail\n\r");
+    }
+}
+
+void AP_UAVCAN::rc_out_sem_give()
+{
+    _rc_out_sem->give();
+}
+
+void AP_UAVCAN::do_cyclic(void)
+{
+
+    uint32_t _servo_bm = SRV_Channels::get_can_servo_bm();
+    uint32_t _esc_bm = SRV_Channels::get_can_esc_bm();
+
+    if (_initialized == true) {
+        auto *node = get_node();
+        const int error = node->spin(uavcan::MonotonicDuration::fromMSec(1));
+        if (error < 0) {
+            hal.scheduler->delay_microseconds(1000);
```

Comment:
> spin will give error if node was not initialized yet

If that's the only time it would give an error then I guess this is fine. Although, like @pavel-kirienko says, it looks like there wouldn't be any issue in just returning here.